Updating copyright headers to match current best practices

tmichalak opened the pull request:
Made applicable changes mentioned in https://github.com/SymbiFlow/ideas/issues/47
* Renamed the `COPYING` file
* Removed `All rights reserved`
* Updated the year
* Added short version of ISC license header to all *.cc and *.h files
* Added short version of ISC license header to all *.py scripts
* Added short version of ISC license header to all bash scripts
* Added short version of ISC license header to all Makefiles
* Added  the SPDX identifier to the license headers

mithro commented on LICENSE:
```diff
@@ -1,4 +1,4 @@
-Copyright (C) 2017  The Project X-Ray Authors. All rights reserved.
+Copyright (C) 2020  The Project X-Ray Authors.
```
`2017-2020`

tmichalak replied:
Done

mithro commented on lib/big_endian_span_test.cc:
```diff
@@ -1,3 +1,18 @@
+/*
```
Let's use the short version of the ISC header.

Also add the SPDX identifier.

tmichalak replied:
@mithro How does a short version look like?

mithro replied:
See first version in https://github.com/SymbiFlow/prjxray/issues/1300

tmichalak replied:
Done

tmichalak commented:
@mithro We also want a license headers in `.py` and `.tcl` files, right? Also with the short version of the ISC header?

mithro commented:
@tmichalak - Any code file, `.py`, `.tcl`, `Makefile`, `.sh`, etc......

aimylios commented on minitests/roi_harness/runme.sh:
```diff
@@ -1,3 +1,12 @@
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
 #!/usr/bin/env bash
```
Duplicate shebang.

tmichalak replied:
Thanks for spotting this. Fixed.

aimylios commented on vagrant/profile.d/vivado.sh:
```diff
@@ -1,3 +1,12 @@
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
 #!/bin/sh
```
Duplicate shebang.

tmichalak replied:
Thanks for spotting this. Fixed.

mithro commented on .github/kokoro/db-full.sh:
```diff
@@ -1,4 +1,13 @@
 #!/bin/bash
+# SPDX-License-Identifier: ISC
```
The SPDX codes go after the license, see the other repository.

tmichalak replied:
I thought I did exactly as I saw in the other repo...
Not the point though. Is such header the desired one?
<pre>
#!/bin/bash
#
# Copyright (C) 2017-2020  The Project X-Ray Authors.
#
# Use of this source code is governed by a ISC-style
# license that can be found in the LICENSE file or at
# https://opensource.org/licenses/ISC
#
# SPDX-License-Identifier: ISC
#
</pre>

tmichalak replied:
@mithro if the header I pasted above is correct, just let me know and will go on with updating them accordingly.

mithro replied:
Extra `#` at the end here.

mithro commented on download-latest-db.sh:
```diff
@@ -1,4 +1,18 @@
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
+# SPDX-License-Identifier: ISC
```
Repeated license here.

tmichalak replied:
Fixed.

tmichalak commented:
@mithro I fixed the bug you mentioned and also added a license check to Travis. The checks currently fail with:
```
ERROR: fasm does not have the LICENSE file.
ERROR: python-sdf-timing does not have the LICENSE file.
```
which shall go away once the licensing PRs for fasm and python-sdf-timing are merged.

mithro commented:
This fixed https://github.com/SymbiFlow/prjxray/issues/1306
